src/Documentation/Advanced-Concepts/Cancellation-Tokens.md

```diff
@@ -0,0 +1,49 @@
+---
+layout: page
+title: Grain cancellation tokens
+---
+
+# Grain cancellation tokens
+
+The Orleans runtime provides mechanism called grain cancellation token, that enable the developer to cancel asynchronous or long-running synchronous operations. 
+
+
+## Description
+**GrainCancellationToken** is a wrapper around standard **.NET System.Threading.CancellationToken**, which enables cooperative cancellation between threads, thread pool work items or Task objects, and can be passed as grain method argument. 
+
+A **GrainCancellationTokenSource** is a object that provides a cancellation token through its Token property and sends a cancellation message by calling its  ``Cancel`` method.  
+
+## Usage
+
+* Instantiate a CancellationTokenSource object, which manages and sends cancellation notification to the individual cancellation tokens.
+``` csharp
+        var tcs = new GrainCancellationTokenSource();
+```
+* Pass the token returned by the GrainCancellationTokenSource.Token property to each grain method that listens for cancellation.
+
+``` csharp
+        var waitTask = grain.LongWait(tcs.Token, TimeSpan.FromSeconds(10));
+```
+* Use the underlying **CancellationToken** property of **GrainCancellationToken** from operations that receive the cancellation token.
```

Comment:
Would it be better to say something like "a cancellable grain operation needs to handle CancellationToken just like it would do in any other .NET code"?